dali/test/python/test_pipeline.py

```diff
@@ -1060,37 +1089,145 @@ def iter_setup(self):
+        super(CachedPipelineTFRecordReader, self).__init__(batch_size, is_cached=is_cached)
+        
+        tfrecord = sorted(glob.glob("/data/imagenet/train-val-tfrecord-480/train-*"))
+        tfrecord_idx = sorted(glob.glob("/data/imagenet/train-val-tfrecord-480.idx/train-*"))
+        self.input = ops.TFRecordReader(path = tfrecord,
+                                        index_path = tfrecord_idx,
+                                        shard_id = 0, 
+                                        num_shards = num_shards, 
+                                        stick_to_shard = True,
+                                        skip_cached_images = skip_cached_images, 
+                                        features = {"image/encoded" : tfrec.FixedLenFeature((), tfrec.string, ""),
+                                                    "image/class/label": tfrec.FixedLenFeature([1], tfrec.int64,  -1)})
+
+    def define_graph(self):
+        inputs = self.input()
+        jpegs = inputs["image/encoded"]
+        labels = inputs["image/class/label"]
+        images = self.decode(jpegs)
+        return (images, labels)
+
+def test_nvjpegdecoder_cached_mxnet_reader():
     batch_size = 26
+    compare_pipelines(CachedPipelineMXNetReader(batch_size, is_cached=False), 
+                      CachedPipelineMXNetReader(batch_size, is_cached=True), 
+                      batch_size=batch_size, N_iterations=20)
 
-    class ComparePipeline(Pipeline):
-        def __init__(self, batch_size=batch_size, num_threads=1, device_id=0, num_gpus=10000):
-            super(ComparePipeline, self).__init__(batch_size, num_threads, device_id, prefetch_queue_depth = 1)
-            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus, stick_to_shard = True)
-            self.decode_non_cached = ops.nvJPEGDecoder(device = "mixed", output_type = types.RGB)
-            self.decode_cached     = ops.nvJPEGDecoder(device = "mixed", output_type = types.RGB,
-                                                       cache_size=8000,
-                                                       cache_threshold=0,
-                                                       cache_type='threshold',
-                                                       cache_debug=False)
+def test_nvjpegdecoder_cached_caffe_reader():
+    batch_size = 26
+    compare_pipelines(CachedPipelineCaffeReader(batch_size, is_cached=False),
+                      CachedPipelineCaffeReader(batch_size, is_cached=True),
+                      batch_size=batch_size, N_iterations=20)
 
-        def define_graph(self):
-            self.jpegs, self.labels = self.input()
-            images_non_cached = self.decode_non_cached(self.jpegs)
-            images_cached = self.decode_cached(self.jpegs)
-            return (images_non_cached, images_cached)
+def test_nvjpegdecoder_cached_caffe2_reader():
```

Comment:
I think you can use https://nose.readthedocs.io/en/latest/writing_tests.html#test-generators as I did in https://github.com/NVIDIA/DALI/blob/master/dali/test/python/test_decoders.py#L50